patches/net/minecraft/client/multiplayer/KnownPacksManager.java.patch

```diff
@@ -0,0 +1,20 @@
+--- a/net/minecraft/client/multiplayer/KnownPacksManager.java
++++ b/net/minecraft/client/multiplayer/KnownPacksManager.java
+@@ -44,9 +_,17 @@
+             }
+         }
+ 
++        if (list.size() > 1024) {
++            list = list.subList(0, 1024);
++            list1 = list1.subList(0, 1024);
++            LOGGER.warn("NeoForge: too many KnownPacks requested; only the first 1024 will be sent via KnownPack, the rest will be synced normally");
```

Comment:
How often will this get called? Inlining a `LogUtils.getLogger()` instead of patching in a field is probably fine.